tf2/include/tf2/time.h

```diff
@@ -52,6 +53,23 @@ namespace tf2
     return std::chrono::system_clock::now();
   }
 
+  inline Duration durationFromSec(double t)
+  {
+    uint32_t sec, nsec;
+    sec = (uint32_t)floor(t);
+    nsec = (uint32_t)std::round((t-sec) * 1e9);
```

Comment:
continued from https://github.com/ros2/geometry2/pull/20#discussion_r113841044 

@wjwwood wrote:
> Have you tried this function with `t_sec` < 0? I think you might want `std::trunc` rather than `std::floor`, and that makes me think that the nanoseconds logic might not be perfect either.
Maybe you could use this function's "possible implementation":
http://en.cppreference.com/w/cpp/chrono/duration/round
It is not available until C++17, but it's possible implementation might work in more cases than this.

I copied this code from ROS 1's implementation: https://github.com/ros/roscpp_core/blob/057f7fa5c36df67757c6cb91c319faa67cefd6e6/rostime/include/ros/time.h#L153..L158 but switched from using `boost::math::round` to `std::round`. the tests there only seem to test positive times  https://github.com/ros/roscpp_core/blob/057f7fa5c36df67757c6cb91c319faa67cefd6e6/rostime/test/time.cpp#L145 - is that something we should change?

I'll add similar tests here